internal/pkg/cli/deploy/workload.go

```diff
@@ -311,40 +353,52 @@ func (d *workloadDeployer) uploadContainerImage(imgBuilderPusher imageBuilderPus
 		return nil, nil
 	}
 	// If it is built from local Dockerfile, build and push to the ECR repo.
-	buildArg, err := buildArgs(d.name, d.imageTag, d.workspacePath, d.mft)
+	buildArg, err := buildArgs(d.name, d.workspacePath, d.runtimeImage, d.mft)
 	if err != nil {
 		return nil, err
 	}
-	digest, err := imgBuilderPusher.BuildAndPush(dockerengine.New(exec.NewCmd()), buildArg)
-	if err != nil {
-		return nil, fmt.Errorf("build and push image: %w", err)
+	digests := make(map[string]RuntimeImage, len(buildArg))
+	for k, v := range buildArg {
+		digest, err := imgBuilderPusher.BuildAndPush(dockerengine.New(exec.NewCmd()), v)
+		if err != nil {
+			return nil, fmt.Errorf("build and push image: %w", err)
+		}
+		digests[k] = RuntimeImage{
+			Digest:         digest,
+			CustomTag:      d.runtimeImage.CustomTag,
+			GitShortCommit: d.runtimeImage.GitShortCommit,
+			UUID:           d.runtimeImage.UUID,
+		}
 	}
-	return aws.String(digest), nil
+	image := digests[d.name]
+	return &image.Digest, nil
```

Comment:
In this PR I am passing only main container `Digest`. In the next refactor the return type will be changed to `map[string]RuntimeImage`.